src/components/side_panel/split_to_columns_panel/split_to_columns_panel.ts

```diff
@@ -0,0 +1,136 @@
+];
+
+interface Props {
+  onCloseSidePanel: () => void;
+}
+
+interface State {
+  separatorValue: SeparatorValue;
+  customSeparator: string;
+  addNewColumns: boolean;
+}
+
+export class SplitIntoColumnsPanel extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-SplitIntoColumnsPanel";
+
+  state = useState<State>({ separatorValue: "auto", addNewColumns: false, customSeparator: "" });
+  errorMessages: string[] = [];
+  warningMessages: string[] = [];
+
+  setup() {
+    onWillUpdateProps(() => {
+      this.errorMessages = this.getErrorMessages();
+      this.warningMessages = this.errorMessages.length === 0 ? this.getWarningMessages() : [];
+      // The feature makes no sense if we are editing a cell, because then the selection isn't active
+      // Stop the edition when the panel is mounted, and close the panel if the user start editing a cell
+      if (this.env.model.getters.getEditionMode() !== "inactive") {
+        this.props.onCloseSidePanel();
+      }
+    });
+
+    onMounted(() => {
+      this.env.model.dispatch("STOP_EDITION");
+    });
+  }
+
+  onSeparatorChange(value: SeparatorValue) {
+    this.state.separatorValue = value;
+  }
+
+  updateCustomSeparator(ev: InputEvent) {
+    if (!ev.target) return;
+    this.state.customSeparator = (ev.target as HTMLInputElement).value;
+  }
+
+  updateAddNewColumnsCheckbox(ev: Event) {
+    if (!ev.target) return;
+    this.state.addNewColumns = (ev.target as HTMLInputElement).checked;
+  }
+
+  confirm() {
+    interactiveSplitToColumns(this.env, this.separatorValue, this.state.addNewColumns);
+  }
+
+  private getErrorMessages(): string[] {
+    const cancelledReasons = this.env.model.canDispatch("SPLIT_TEXT_INTO_COLUMNS", {
+      separator: this.separatorValue,
+      addNewColumns: this.state.addNewColumns,
+      force: true,
+    }).reasons;
+
+    const errors = new Set<string>();
+
+    for (const reason of cancelledReasons) {
+      switch (reason) {
+        case CommandResult.SplitWillOverwriteContent:
+        case CommandResult.EmptySplitSeparator:
+          break;
+        default:
+          errors.add(SplitToColumnsTerms.Errors[reason] || SplitToColumnsTerms.Errors.Unexpected);
+      }
+    }
+    return Array.from(errors);
+  }
+
+  private getWarningMessages(): string[] {
+    const warnings: string[] = [];
+    const cancelledReasons = this.env.model.canDispatch("SPLIT_TEXT_INTO_COLUMNS", {
+      separator: this.separatorValue,
+      addNewColumns: this.state.addNewColumns,
+      force: false,
+    }).reasons;
+    if (cancelledReasons.includes(CommandResult.SplitWillOverwriteContent)) {
+      warnings.push(SplitToColumnsTerms.Errors[CommandResult.SplitWillOverwriteContent]);
+    }
+    if (
+      this.state.separatorValue !== "auto" &&
+      !this.env.model.getters.isSeparatorInSelection(this.separatorValue)
+    ) {
+      warnings.push(SplitToColumnsTerms.Errors.NoSeparatorMatch);
```

Comment:
this specific warning is kind of strange since it is not the result of a 'canDispatch' and therefore, the command would not be rejected. I can keep clicking on the 'confirm' button which will still dispatch stuff.

Reply:
Changed it into an error and added it in the allowDispatch